### ISSUE TITLE
Error installing Joomla 3.6.2 on a local server when MySQL user root has non-empty password.

### ISSUE THREAD

**Issue by nvyush** [joomla/joomla-cms#11537]
#### Steps to reproduce the issue

On local server:
1) set non-empty password for MySQL user 'root';
2) prepare Joomla for installation (unpack files and create database);
3) start installation (open '***/installation/index.php' page).
#### Expected result

The first page of Joomla! install wizard.
#### Actual result

Eror message: 'JDatabaseExceptionConnecting: Could not connect to MySQL. in ***\libraries\joomla\database\driver\mysqli.php on line 183'
#### System information (as much as possible)

Joomla! installation package from https://github.com/joomla/joomla-cms/releases/download/3.6.2/Joomla_3.6.2-Stable-Full_Package.zip

Database Version:   5.5.5-10.1.13-MariaDB
Database Collation:     utf8mb4_general_ci
Database Connection Collation:  utf8mb4_general_ci
PHP Version:    5.6.19
Web Server:     Apache
WebServer to PHP Interface:     apache2handler
Joomla! Version:    Joomla! 3.6.2 Stable [ Noether ] 4-August-2016 23:41 GMT
Joomla! Platform Version:   Joomla Platform 13.1.0 Stable [ Curiosity ] 24-Apr-2013 00:00 GMT
User Agent:     Mozilla/5.0 (Windows NT 10.0; WOW64; rv:47.0) Gecko/20100101 Firefox/47.0

**Comment by smz** [comment]
I can't confirm this (_but I'm on PHP 7.0.9, MySQL 5.6.31_): I'm presented with the usual first page and then second page (see snapshot)

But wait... I executed /index.php, *not* **/installation/**index.php to start installation: I guess this could be your mistake...

![capture](https://cloud.githubusercontent.com/assets/552475/17567093/429354f0-5f3e-11e6-89b9-9bf96e21fcad.PNG)

**Comment by brianteeman** [comment]
Sergio check your address bar and you will see you might have only typed
/index.php but you ended up at installation/index.php

**Comment by smz** [comment]
correct: I was about posting that even with /installation/index.php (_directly entered_), i'm not affected by this issue...

**Comment by ot2sen** [comment]
Have successfully created several 3.6.2 sites locally this week, on a much
similar setup.
Main difference in setup from reporter seems to be I am using MySQL while
they are using the fork MariaDB.

*Databaseversion* 5.6.24
*Databasesammenkøring* utf8_general_ci
*Database forbindelse kollation* utf8mb4_general_ci
*PHP version* 5.6.8
*Webserver* Apache/2.4.12 (Win32) OpenSSL/1.0.1l PHP/5.6.8 mod_wsgi/3.5
Python/2.7.9
*WebServer til PHP interface* apache2handler
*Joomla!-version* Joomla! 3.6.2 Stable [ Noether ] 4-August-2016 23:41 GMT
*Joomla! platformsversion* Joomla Platform 13.1.0 Stable [ Curiosity ]
24-Apr-2013 00:00 GMT

*Brugeragent*



Mozilla/5.0 (Windows NT 10.0; WOW64; Trident/7.0; rv:11.0) like Gecko


Have you succeeded in creating any sites locally with that new root user
with a non-empty password?
Thinking you could be having more root users now where the new not having
all privileges granted.
MariaDB generally is reported to work with former releases, dunno if there
could be an issue for 3.6.2 in particular.

**Comment by smz** [comment]
> MariaDB generally is reported to work with former releases, dunno if there
could be an issue for 3.6.2 in particular.

It perfectly works for me on my **production** sites where I have MariaDB 5.5.47 under CentOS 7.2 and recently **upgraded** most of them from J! 3.4.8 to J! 3.6.2. No new install over there, thus.

If needed, I can try.

**Comment by ggppdk** [comment]
Tested with the same DB version and on windows 10 as the original report
```
Database Version: 5.5.5-10.1.13-MariaDB  (xampp distribution)
```

but not with PHP 5.6.19 instead i have PHP 7.0.8 and it works

**Comment by nvyush** [comment]
When password for MySQL user 'root' is non-empty:
![error-install](https://cloud.githubusercontent.com/assets/20832832/17581548/22b18672-5faf-11e6-92ab-96c6a7a67324.png)

When password for MySQL user 'root' is empty:
![correct-installing](https://cloud.githubusercontent.com/assets/20832832/17581597/7081f6de-5faf-11e6-9c6c-129aa88c65ab.png)

Perhaps the problem is the server. I try to reinstall it.

**Comment by nvyush** [comment]
I tried to install old versions of Joomla! when the xdebug is on. 3.5.1 - the same error, 3.4.8 - no errors.

**Comment by smz** [comment]
I made a quick check of the differences in installation between 3.4.8 and 3.5.1.

I only see something that jumps to my eyes (_because of that "Failed to start the session..." error you get_): 

In 3.4.8 we had `JFactory::$session = null;` in the constructors of the following classes and it has been removed in 3.5.1:
- InstallationControllerSetdefaultlanguage
- InstallationControllerInstallEmail
- InstallationControllerInstallLanguages

I don't know if this has anything to do with the issue at hand, but I think it can be a "lead" for someone more experienced than I am.

The other thing that probably has changed is that **maybe** somewhere we make a check of the the DB connection collation. Could it be the case that this check is somehow triggered even before having provided the DB parameters??

Just two 2 cents from my part without presuming I exactly know what I'm talking about...
